core/modules/contextual/js/contextual.js

```diff
@@ -23,23 +23,110 @@ Backdrop.behaviors.contextualLinks = {
+      allWrappers.forEach(function(info) {
+        info.wrapper.css('margin', '0');
+      });
+
+      // Recalculate margins to avoid collisions.
+      var dir = $('html').attr('dir');
+      const hSize = 28; // width of trigger wrapper
+      const vSize = 19; // height of trigger wrapper
+      var n = allWrappers.length;
+      for (let i = 0; i < n; i++) {
+        var follower = allWrappers[i];
+        // Compare follower against all of its predecessors in the list (any of
+        // which may have already been adjusted).
+        for (let j = 0; j < i; j++) {
+          var leader = allWrappers[j];
+          // Adjust the position of follower if necessary to avoid collision
+          // with leader.
+          var leaderOffset = leader.wrapper.offset();
+          var followerOffset = follower.wrapper.offset();
+          // Check vertical overlap.
+          if (!(followerOffset.top >= leaderOffset.top && followerOffset.top < leaderOffset.top + vSize)) {
+            continue;
+          }
+          if (dir == 'ltr') {
+            // Check horizontal overlap.
+            if (followerOffset.left >= leaderOffset.left - hSize && followerOffset.left < leaderOffset.left + hSize) {
+              // We have a collision; shift the follower down if there's room,
+              // otherwise left.
+              if (followerOffset.top + 2 * vSize <= follower.regionOffsetBottom) {
+                // Shift down
+                follower.vShift += vSize;
+                follower.wrapper.css('margin-top', follower.vShift);
+              }
+              else {
+                // Shift left and start a new column.
+                follower.vShift = 0;
+                follower.hShift += hSize;
+                follower.wrapper.css('margin-top', follower.vShift);
+                follower.wrapper.css('margin-right', follower.hShift);
+              }
+            }
+          }
+          else { // rtl
+            // Check horizontal overlap.
+            if (followerOffset.left > leaderOffset.left - hSize && followerOffset.left <= leaderOffset.left + hSize) {
+              // We have a collision; shift the follower down if there's room,
+              // otherwise right.
+              if (followerOffset.top + 2 * vSize <= follower.regionOffsetBottom) {
+                // Shift down
+                follower.vShift += vSize;
+                follower.wrapper.css('margin-top', follower.vShift);
+              }
+              else {
+                // Shift right and start a new column.
+                follower.vShift = 0;
+                follower.hShift += hSize;
+                follower.wrapper.css('margin-top', follower.vShift);
+                follower.wrapper.css('margin-left', follower.hShift);
+              }
+            }
+          }
+        }
+      }
+    }
+    $(document).ready(adjustContextualLinks);
+    Backdrop.optimizedResize.add(adjustContextualLinks, 'adjustContextualLinks');
```

Comment:
In testing, I found that optimizedResize() fires a lot more often than I would have thought. It also fires on scroll, which we don't want for something like this. So instead a debounce with some documentation might be the best course:
 
```suggestion

    // Usually Backdrop.optimizedResize() would be used for a window resize
    // event, but this potentially expensive operation should be limited to
    // firing infrequently, so Backdrop.debounce() is used here instead.
    $(window).on('resize', Backdrop.debounce(adjustContextualLinks, 500));
```